src/router.js

```diff
@@ -186,7 +186,7 @@ export class Router {
   * @param params The route params to be used to populate the route pattern.
   * @returns {string} A string containing the generated URL fragment.
   */
-  generate(name: string, params?: any): string {
+  generate(name: string, params?: any, absolute?: boolean = false): string {
```

Comment:
is definitely a good idea to use an option object, it'll make keeping BC changes easier